services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -0,0 +1,419 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.services;
+
+import com.linkedin.kmf.services.configs.TopicManagementConfig;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import kafka.admin.AdminUtils;
+import kafka.admin.PreferredReplicaLeaderElectionCommand;
+import kafka.cluster.Broker;
+import kafka.common.TopicAndPartition;
+import kafka.utils.ZkUtils;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.security.JaasUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.kmf.common.Utils.ZK_CONNECTION_TIMEOUT_MS;
+import static com.linkedin.kmf.common.Utils.ZK_SESSION_TIMEOUT_MS;
+
+import scala.collection.Seq;
+
+
+/**
+ * Runs this periodically to rebalance the monitored topic across brokers and to reassign elected leaders to brokers so that the
+ * monitored topic is sampling all the brokers evenly.
+ */
+public class TopicManagementService implements Service  {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TopicManagementService.class);
+
+  /**
+   * The state of the monitored topic.
+   */
+  static class TopicState {
```

Comment:
TopicState is a logically separate concept from the thing that is doing the monitoring and rebalancing.  If it is up to me then I would rather leave it untouched.